library/src/updater.rs

```diff
@@ -370,33 +370,34 @@ pub fn current_boot_patch() -> anyhow::Result<Option<PatchInfo>> {
 }
 
 pub fn report_launch_start() -> anyhow::Result<()> {
-    with_config(|config| {
-        let mut state =
-            UpdaterState::load_or_new_on_error(&config.storage_dir, &config.release_version);
-        // Validate that we have an installed patch.
-        // Make that patch the "booted" patch.
-        state.activate_current_patch()?;
-        state.save()
-    })
+    // We previously set the "current" patch the value of the "next" patch, but no longer
+    // do so because the semantics have changed:
+    //   current is now "last successfully booted patch"
+    //   next is now "patch to boot next"
+
+    Ok(())
 }
 
 /// Report that the current active path failed to launch.
 /// This will mark the patch as bad and activate the next best patch.
 pub fn report_launch_failure() -> anyhow::Result<()> {
     info!("Reporting failed launch.");
+
     with_config(|config| {
         let mut state =
             UpdaterState::load_or_new_on_error(&config.storage_dir, &config.release_version);
 
+        // Attempting to get next_boot_patch might return None if the failure was due to
+        // the patch being altered on disk.
```

Comment:
Made a decision not to report patches that fail due to local disk changes. This felt reasonable to me, as there is nothing the app developer can do in this case.

Reply:
I guess it comes down to how it manifests (does the user notice?).  A patch failure is still a patch failure and probably worth reporting, possibly with context of some form?

Reply:
Agreed, will update